File: RailsProject/app/models/notification.rb
# The model of the object Notification
# Contain the relation and the validation
# Can provide some features linked to this model
#
# ==== Attributes
#
# - +id+ - (integer) - The ID of the object
# - +user_id+ - (integer) - The ID of the user
# - +link+ - (string) - The link to see what about the notification is
# - +description+ - (string) - The text inside the notification
#
# ==== Associations
#
# - +belongs_to+ - :user
#
class Notification < ActiveRecord::Base
  belongs_to :user
  belongs_to :from, class_name: "User", foreign_key: 'from_user_id'

  validates :user, :notif_type, :from, presence: true
	validates_inclusion_of :read, :in => [true, false]

  # The strong parameters to save or update object
  def self.notification_params(parameters)
    parameters.require(:notification).permit(:user_id, :link, :description)
  end

  # Filter of information for the API
  #
  # Fields returned : [:id, :title, :date]
  def self.miniKey
    [:id, :created_at, :notif_type, :read, :link]
  end
end

File: W8_SoonZik/SoonZik/Models/Notification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoonZik.Models
{
    class Notification
    {
        public int id { get; set; }
        public int user_id { get; set; }
        public string notif_type { get; set; }
        public int from_user_id { get; set; }
        public bool read { get; set; }
        public string link { get; set; }
    }
}
